File: terre06.rb
# Inverser une chaîne

argument = ARGV                


if argument.length != 1                                  # si l'utilisateur n'entre rien ou plus d'un argument
    print "erreur : insérer un argument"                 # affiche ce message d'erreur
else

    chaine = ARGV[0].chars                               # l'argument est stockée en caractères individuels dans la liste chaine
    index = chaine.length - 1                            # variable contenant l'index du dernier élément de la liste
    
    while   index >= 0  do                               # sinon, tant que l'index est supérieur ou égal à 0 (premier élément du tableau)
        print chaine[index]                              # affiche le dernier élément de la liste chaine
        index -= 1                                       # affiche l'élément précédant le dernier élément de la liste chaine
    end 
end

print "\n"

File: terre06.js
// Inverser une chaîne

let arguments = process.argv;

let string = process.argv[2];

if (arguments.length > 3) {
	console.log("erreur : n'insérez qu'un argument");
}
else {
	reversedString = "";
	for (let i = string.length - 1; i >= 0; i --){
		reversedString += string[i];
	}
	console.log(reversedString);
}